Smarty/plugins/modifier.escape.php

```diff
@@ -23,7 +23,7 @@ function smarty_modifier_escape($string, $esc_type = 'html', $char_set = 'ISO-88
 {
     switch ($esc_type) {
         case 'html':
-            return htmlspecialchars($string, ENT_QUOTES, $char_set);
+            return htmlspecialchars($string ?: '', ENT_QUOTES, $char_set);
```

Comment:
Mixed thoughts:
* Upstream does not address this and lets it error the same as now.
* There's the time savings from doing it here, but possibly hides some code bug. (Although of all the stricter php warnings I think this "null should be a string" one is the least useful one.)
* If we are going to do it here, then should the whole function just return early with `''` if $string is null?

Reply:
So this issue is kind of making civi unusable if you have your settings turned up to 11. I think I'd be ok to merge a version of it that is like the third bullet above. And while it's not in v3 of smarty, looking at the latest master it does this: `$string = (string)$string;`, so we could do the same:

```diff
 */
function smarty_modifier_escape($string, $esc_type = 'html', $char_set = 'ISO-8859-1')
{
+  $string = (string)$string;
    switch ($esc_type) {
        case 'html':
            return htmlspecialchars($string, ENT_QUOTES, $char_set);
```

although this seems more efficient when it's null:

```diff
 */
function smarty_modifier_escape($string, $esc_type = 'html', $char_set = 'ISO-8859-1')
{
+  if ($string === NULL) {
+    return '';
+  }
    switch ($esc_type) {
        case 'html':
            return htmlspecialchars($string, ENT_QUOTES, $char_set);
```